api/views/event.py

```diff
@@ -28,3 +29,27 @@ def get_event_types(self, request, pk=None):
         Returns a dictionary of the defined event types
         """
         return Response(EVENT_TYPE_CHOICES)
+
+    @action(detail=False, methods=['post'], url_path="duplicate-event")
+    def duplicate_event(self, request):
+        """
+        Duplicates an event as well as the questions within the event.
+        """
+        event = Event.objects.filter(id=request.data.get("event", None)).first()
+        course = CanvasCourse.objects.filter(id=request.data.get("course", None)).first()
+        old_event_id = event.id
+        event.id = None
+        event.course = course
+        event.name += ' (Copy)'
+        event.save()
+        for question in Question.objects.all().filter(event=old_event_id):
+            question.id = None
+            question.pk = None
+            question.question_ptr_id = None
```

Comment:
I'm like 99.9% sure that we need to set the `question.variablequestion_ptr_id = None` too. But maybe because `VariableQuestion` is a child of `Question` this could work. But let's be safe and do it properly.

Reply:
Okay I will set this as well.